### ISSUE TITLE
expm is allocating

### ISSUE THREAD

**Issue by nlw0** [JuliaArrays/StaticArrays.jl#578]
It seems this line from the exp() method
https://github.com/JuliaArrays/StaticArrays.jl/blob/786b6f006bd9fcbf7445e6bef018a666f88a44da/src/expm.jl#L71

is causing a couple of allocations, what I imagine is not something desirable. Just using opnorm(A,1) prevents it.

I am not sure how to go about making a PR, should I go for a non-allocating version of this, or just move back to opnorm?

**Comment by nlw0** [comment]
#579

**Comment by c42f** [comment]
The underlying problem here seems to be that `sum(A; dims=Val(1))` is allocating:

```julia
julia> A = @SArray rand(4,4);

julia> @btime sum($A; dims=Val(1))
  28.764 ns (2 allocations: 192 bytes)
1×4 SArray{Tuple{1,4},Float64,2,4} with indices SOneTo(1)×SOneTo(4):
 1.09627  1.93056  1.62516  2.25825
```

**Comment by c42f** [comment]
Ultimately root cause may be a dup of #540 and #541

**Comment by mateuszbaran** [comment]
It's fixed on master (most likely a duplicate of #540).
